pkg/controllers/networkpolicy/policy_controller.go

```diff
@@ -238,8 +238,8 @@ func (c *policyController) syncToDatastore(key string) error {
 		}
 
 		clog.Infof("Update NetworkPolicy in Calico datastore with resource version %s", p.ResourceVersion)
-		p.ResourceVersion = gp.ResourceVersion
-		_, err = c.calicoClient.NetworkPolicies().Update(c.ctx, &p, options.SetOptions{})
+		gp.Spec = p.Spec
```

Comment:
Do we care about not copying the annotations or labels?

Reply:
yeah good point, we probably do want to do that.

Reply:
On second thought, I'm not so sure.


It's also a larger change than it seems and requires a new libcalico, so I'm not going to do it in this PR.